themes/bootstrap5/scss/components/search.scss

```diff
@@ -654,6 +654,11 @@ header .container.navbar { margin-bottom: 0; }
     height: auto !important;
   }
 }
+@include color-mode(dark, true) {
+  .top-title {
+    color: #fff;
+  }
+}
```

Comment:
This would be an example of overriding a locally defined color, in this case for `.top-title`.  It works fine.  

It would be more elegant IMHO to put this `@include` within the main styles for `.top-title` above.  But unfortunately, supporting that kind of non-root-level media query causes problems with some of the built-in Bootstrap styles.  Also, since we handle different breakpoints as top-level `@include`s just above here, I don't see any harm in handling color mode the same way.